src/http/body.rs

```diff
@@ -8,7 +8,31 @@ use tokio_core::io::{Codec, EasyBuf};
 
 use super::Chunk;
 
+pub struct Chunked {}
+
+impl Codec for Chunked {
+    type In = Chunk;
+    type Out = Chunk;
+
+    fn decode(&mut self, buf: &mut EasyBuf) -> io::Result<Option<Self::In>> {
+        let len = buf.len();
+        trace!("Chunked decode {} bytes", len);
+        let body = buf.drain_to(len);
+        Ok(
+            Some(
+                Chunk(Vec::from(body.as_ref()))
+            )
+        )
```

Comment:
That being said, this is the logic for the case when there is no content length or transfer encoding chunked header.